core/forkid/forkid.go

```diff
@@ -297,5 +292,9 @@ func gatherForks(config *params.ChainConfig) ([]uint64, []uint64) {
 	if len(forksByTime) > 0 && forksByTime[0] == 0 {
 		forksByTime = forksByTime[1:]
 	}
+	// Skip any forks by time that are non-zero, but before genesis.
+	if len(forksByTime) > 0 && forksByTime[0] <= genesis {
+		forksByTime = forksByTime[1:]
+	}
```

Comment:
You can have any number of forks placed before the genesis if the genesis has nonzero time. so this needs to be a loop. Something like: 
```suggestion
	while len(forksByTime) > 0 && forksByTime[0] <= genesis {
		forksByTime = forksByTime[1:]
	}
```

Reply:
Ah good point. Fixed.